pkg/api/api.go

```diff
@@ -58,10 +59,16 @@ type KeyHandleReader interface {
 	Export(keyID string) ([]byte, kms.KeyType, error)
 }
 
-// DIDCreator defines DID creation APIs.
+// CreateDIDOpts represents the various options for the DIDCreator.Create method.
+type CreateDIDOpts struct {
+	DIDDoc *did.Doc
```

Comment:
What is this DIDDoc option?

Reply:
The various Aries DID creator methods (which I call from wallet-sdk here) require this as an input parameter (see https://github.com/hyperledger/aries-framework-go/blob/a755fd87d699b886341536c690dd78616cf623bd/pkg/vdr/key/creator.go#L31 for reference).

Is seems the Aries DID creator methods use the fields from that DID Doc input parameter as options, essentially. Not sure why they work like that. Maybe @Baha-sk @fqutishat have some idea?

Perhaps I'll have to define those "options" as input parameters for our wallet-sdk function, and then inside our wallet-sdk function I can translate them to the "DID Doc" input parameter?

Reply:
delete it, we dont need it here. In aries we follow some opensource specs & some protocols where we had to use some custom data types or interfaces.

Try to keep wallet SDK as simple as possible